templates/zerver/emails/password_reset.subject

```diff
@@ -1 +1,3 @@
-Zulip password reset
+{% if attempted_realm %} Failed password reset attempt
+{% else %} Somebody requested a new password for your Zulip account
+{% endif %}
```

Comment:
Just read the email we sent in the attempted_realm case; I think both cases should be "Somebody requested a new password for your Zulip account", actually.
The three cases are:
1. You reset your password and are coming to check your email, waiting for an email from Zulip
2. You didn't do anything and suddenly get this email
3. You tried to reset your password for the wrong realm

For (1), I think it doesn't matter what the subject line is, as long as it contains the word "password"; you're not going to read it.
I think we're optimizing for (2) and (3), which is where we want to be the most sensitive and/or kind.

(rebased and pushed, if you agree with the changes)